classes/booking_manager.php

```diff
@@ -273,6 +290,35 @@ public function validate($timenow = null): array {
                     new lang_string('error:invalidstatusspecified', 'mod_facetoface', $entry->status),
                 ];
             }
+
+            if (isset($userid)) {
+                if (!isset($userrows[$userid])) {
+                    $userrows[$userid] = [];
+                }
+                if (!isset($usersessions[$userid])) {
+                    $usersessions[$userid] = [];
+                }
+
+                $userrows[$userid][$row] = $entry->email;
+
+                if ($session && !in_array($session->id, $usersessions[$userid])) {
+                    $usersessions[$userid][] = $session->id;
+                }
+            }
+        }
+
+        if ($this->facetoface->signuptype != MOD_FACETOFACE_SIGNUP_MULTIPLE) {
+            foreach ($usersessions as $userid => $sessions) {
```

Comment:
Potential refactor idea, that IMO would be more readable:

```
$doublebookedsessions = array_filter($usersessions, fn($userid, $sessions) => return count($sessions) > 1, ARRAY_FILTER_USE_BOTH)
$doublebookederrors = array_map(fn($userid) => [...build error here], $doublebookedsessions)
$errors = array_merge($errors, $doublebookederrors);
```

Reply:
I've changed this a bit along with the `$usersessions` array structure so it matches how the over capacity sessions validation works. Let me know what you think.